sgt-back-end/index.js

```diff
@@ -0,0 +1,137 @@
+/* eslint-disable no-console */
+
+const pg = require('pg');
+const express = require('express');
+const app = express();
+
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:lfz@localhost/studentGradeTable'
+});
+
+app.get('/api/grades', (req, res, next) => {
+
+  const sql = `
+    select *
+    from "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+      const grade = result.rows;
+      res.status(200).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res, next) => {
+  const data = req.body;
+
+  if (data.name === '' || data.course === '' || data.score === '' || data.score < 0 || data.score > 100) {
+    res.status(400).json({
+      error: 'Invalid or missing information. Please try again.'
+    });
+    return;
+  }
+
+  const sql = `
+    insert into "grades" ("name", "course", "score")
+    values($1, $2, $3)
+    returning *`;
+
+  const values = [data.name, data.course, data.score];
+
+  db.query(sql, values)
+    .then(result => {
+      const grade = result.rows[0];
+      res.status(201).json(grade);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occurred.'
+      });
+    });
+});
+
+app.put('/api/grades/:gradeId', (req, res, next) => {
+  const data = req.body;
+  const gradeId = parseInt(req.params.gradeId, 10);
+
+  if (!Number.isInteger(gradeId) || gradeId <= 0 || data.name === '' || data.course === '' || data.score === '' || data.score < 0 || data.score > 100) {
```

Comment:
As mentioned above I would break this into a few different `if` statements. This keeps the code more readable, easier to debug, and gives you the ability to give the client more concise error messages.